s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/config/S3SourceConfig.java

```diff
@@ -340,8 +339,8 @@ public String getAwsS3BucketName() {
         return getString(AWS_S3_BUCKET_NAME_CONFIG);
     }
 
-    public OutputFormat getOutputFormat() {
-        return OutputFormat.valueOf(getString(OUTPUT_FORMAT_KEY).toUpperCase(Locale.ROOT));
+    public InputFormat getInputFormat() {
+        return InputFormat.valueOf(getString(INPUT_FORMAT_KEY).toUpperCase(Locale.ROOT));
     }
 
     Region getAwsS3Region() {
```

Comment:
I don't understand this connector's partitioning strategy and multi-task strategy.

It looks like the taskConfigs method generates N identical tasks, which will all try to write all objects, causing lots of duplicates.

It looks like the user is expected to configure "topic.partitions", and configure N different connectors to read each partition of the data? That's very unusual.

Also topic.partitions limits the partitions that are loaded by the OffsetManager, but doesn't appear to limit which partitions actually get transferred. Maybe it does, but I couldn't figure out how.

Reply:
Totally agree with this

Reply:
Agree and we were doubting this, and we would like to introduce the functionality about topic partitioning strategy in a different jira.

Reply:
Related ticket https://aiven.atlassian.net/browse/KCON-27